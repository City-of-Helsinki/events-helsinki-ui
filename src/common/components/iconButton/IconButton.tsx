import classNames from 'classnames';
import React from 'react';

import styles from './iconButton.module.scss';

interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  ariaLabel: string;
  icon: React.ReactElement;
  size: 'default' | 'small';
}

const IconButton: React.FC<Props> = ({
  ariaLabel,
  icon,
  size,
  type = 'button',
  ...rest
}) => {
  return (
    <button
      aria-label={ariaLabel}
      className={classNames(styles.iconButton, styles[size])}
      {...rest}
    >
      {icon}
    </button>
  );
};

export default IconButton;
