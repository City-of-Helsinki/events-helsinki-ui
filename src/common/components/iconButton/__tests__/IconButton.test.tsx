import { IconAngleRight } from 'hds-react';
import * as React from 'react';
import renderer from 'react-test-renderer';

import IconButton from '../IconButton';

test('IconButton matches snapshot', () => {
  const component = renderer.create(
    <IconButton ariaLabel="Aria label" icon={<IconAngleRight />} size="small" />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

export {};
