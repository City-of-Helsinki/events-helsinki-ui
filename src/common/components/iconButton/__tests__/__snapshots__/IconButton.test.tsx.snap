// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton matches snapshot 1`] = `
<button
  aria-label="Aria label"
  className="iconButton small"
>
  <svg
    className="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
    style={Object {}}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fillRule="evenodd"
    >
      <path
        d="M0 24V0h24v24z"
      />
      <path
        d="M13 12.5l-5-5L9.5 6l6.5 6.5L9.5 19 8 17.5z"
        fill="currentColor"
      />
    </g>
  </svg>
</button>
`;
